pwnlib/crypto/ciphers/vigenere.py

```diff
@@ -0,0 +1,80 @@
+import string
+
+class CipherVigenere:
+    def __init__(self):
+        self.alphabet = string.ascii_lowercase
+
+
+    def new_alph(self, char):
+        char = char.lower()
+        new_alph = self.alphabet[self.alphabet.index(char):] + self.alphabet[:self.alphabet.index(char)]
+        return new_alph
+
+
+    def encrypt(self, cleartext, key):
+        if key is None:
+            raise ValueError('No key given')
+
+        if not (type(key) is str):
+            if(type(key) is int):
+                raise ValueError('Key must be string not int')
+            else:
+                raise ValueError('Key needs to be string')
+
+        if(len(key) < len(cleartext)):
+            raise ValueError('Key shorter then cleartext - loosing data')
+
+        ciphertext = ''
+
+        i = 1
+        for char in key:
+            new = self.new_alph(char)
+            for t in cleartext:
+                if self.alphabet.count(t) == 1 :
+                    ciphertext += new[self.alphabet.index(t)]
+                    cleartext = cleartext[i:]
+                    break
+                elif self.alphabet.count(t.lower()) == 1:
+                    ciphertext += new[self.alphabet.index(t.lower())].upper()
+                    cleartext = cleartext[i:]
+                    break
+                else:
+                    ciphertext += t
+                    cleartext = cleartext[i:]
+                    break
+                i += 1
+        return ciphertext
+
+
+    def decrypt(self, ciphertext, key):
+        if(key is None):
+            raise ValueError('No key given')
+
+        if not (type(key) is str):
```

Comment:
We should probably handle bytes() everywhere that we handle str(), for Python3-related reasons.  It's reasonable for a key to be a `bytes` object and not a `str` object.

Reply:
Also, use `isinstance`, not `type`.  It's reasonable to work with subclasses of `bytes` or `str`.  `six.ensure_bytes` is probably useful here.

Reply:
Can you give me a hint on how to change the code to what is wanted? Its my first pull request to a project and i am stumbling on some parts.